### PR TITLE
RecyclerView list news

### PR DESCRIPTION
**Pull request by camilo05**
Screenshot_1632434295

Sep-23-2021 14-03-03

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 1]
```diff
@@ -1,18 +1,86 @@
...
+            presenter.onAddItemsNews(news)
+            adapter = NewsAdapter(presenter)
+            recyclerview.adapter = adapter
+        }
+    }
+
+    override fun setListeners() {
+        with(binding) {
+            swipe.setOnRefreshListener {
+                Handler().postDelayed({
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt:68]: Nice! Remember to delete this when implementating the integration

File: app/src/main/java/ar/com/wolox/android/example/utils/ViewExtensions.kt [hunk 2]
```diff
@@ -14,4 +21,10 @@ fun View.toggleVisibility(visible: Boolean) {
  */
 fun View.togglePresence(present: Boolean) {
     visibility = if (present) View.VISIBLE else View.GONE
-}
\ No newline at end of file
+}
+
+fun String.abbreviationDayFormat(): String {
+    return PrettyTime().format(Date(SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").parse(this).time))
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/utils/ViewExtensions.kt]: Move this string to a constant

File: app/src/main/java/ar/com/wolox/android/example/utils/ViewExtensions.kt [hunk 3]
```diff
@@ -14,4 +21,10 @@ fun View.toggleVisibility(visible: Boolean) {
...
+}
+
+fun ImageView.glideImage(url: String) = Glide.with(this).load(url).placeholder(ColorDrawable(Color.WHITE)).into(this)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/utils/ViewExtensions.kt]: Excellent!
Glide also provides the .error() method, so you can set an image to be rendered when the url has an error.

File: app/src/main/res/layout/news_item.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="@dimen/static_spacing_largest"
+            android:layout_marginRight="@dimen/static_spacing_largest"
+            android:textColor="@android:color/black"
+            app:layout_constraintBottom_toTopOf="@id/description"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:textSize="18sp"
```
Comment (ffasitella) [inline app/src/main/res/layout/news_item.xml]: Hardcoded dimen

**Review by ffasitella — CHANGES_REQUESTED** [review]
Well done! Left some feedback

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/adapter/NewsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemNewViewHolder {
+        val binding: NewsItemBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.news_item, parent, false)
+        return ItemNewViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ItemNewViewHolder, position: Int) {
+        presenter.onBindNewsView(holder, position)
+    }
+
+    override fun getItemCount(): Int = presenter.getNewsRows()
```
Comment (LisandroDiMeo) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/adapter/NewsAdapter.kt]: Im not sure if its okay to have the presenter here. Usually the adapter only receives a *Collection* of elements, which will use to populate the recycler, and other parameters such like *Listeners*. In your case, passing the list of news to news adapter, will result in having the implementation of getItemCount with something like newsList.size.

**Review by LisandroDiMeo — CHANGES_REQUESTED** [review]
Just one comment! 💪

**Review by rodrigo103 — APPROVED** [review]
Great!!!

**Review by LisandroDiMeo — APPROVED** [review]
Great 👏

**Review by ffasitella — APPROVED** [review]
🙌🏽 ✨

**Pull request merged**